url-utils.js

```diff
@@ -0,0 +1,29 @@
+if (global.DOMParser !== undefined) {
+  // We're in Firefox
```

Comment:
What you've done does work, but because of the configuration you added to webpack you should actually be able to just do this for both:

```js
  const {URL} = require("url");
  module.exports = {
    makeUrlAbsolute(base, relative) {
      return new URL(relative, base).href;
    },
    parseUrl(url) {
      return new URL(url).host;
    }
};
```